include/soci/type-conversion-traits.h

```diff
@@ -36,6 +36,29 @@ struct type_conversion
     }
 };
 
+// Most majority of database engines ignore unsigned integer related types.
+// This hack allows to trait these types as signed integers in the database.
+template<>
+struct type_conversion<unsigned long long>
+{
+    typedef long long base_type;
+
+    static void from_base(base_type const & in, indicator ind, unsigned long long & out)
+    {
+        if (ind == i_null)
+        {
+            throw soci_error("Null value not allowed for this type");
```

Comment:
Is this really correct? A database column of `bigint` type could store nulls perfectly well...

Reply:
Yes, this changes is not acceptable for core.